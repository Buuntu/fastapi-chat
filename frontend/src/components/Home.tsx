import React, { FC, useState } from 'react';
import {
  Button, TextField, Grid,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useHistory } from 'react-router-dom';

import { BACKEND_URL } from '../config';

const useStyles = makeStyles({
  button: {
    marginTop: '20px',
  },
  roomInput: {
    font: 'white',
  },
});

const Home:FC = () => {
  const classes = useStyles();
  const [roomNumber, setRoomNumber] = useState<string>('');
  const history = useHistory();

  const onClick = () => {
    history.push(`room/${roomNumber}`);
  };

  return (
    <div>
      <Grid item xs={12}>
        <TextField
          className={classes.roomInput}
          label="Room Number"
          value={roomNumber}
          onChange={(e) => setRoomNumber(e.currentTarget.value)}
          variant="filled"
        />
      </Grid>
      <Grid item xs={12}>
        <Button
          variant="contained"
          className={classes.button}
          onClick={() => onClick()}
        >
          Create
        </Button>
      </Grid>
    </div>
  );
};

export default Home;
