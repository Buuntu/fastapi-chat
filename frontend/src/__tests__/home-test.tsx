import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';

it('Home renders correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});