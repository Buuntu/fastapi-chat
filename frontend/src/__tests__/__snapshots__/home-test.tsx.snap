// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      onClick={[Function]}
    >
      Click to make request to backend
    </a>
    
    
  </header>
</div>
`;
